src/main/java/org/graylog/plugins/usagestatistics/collectors/NodeCollector.java

```diff
@@ -108,10 +108,20 @@ private NodeInfo buildNodeInfo() {
         return NodeInfo.create(
                 NodeRole.fromCapabilities(serverStatus),
                 Version.CURRENT_CLASSPATH.toString(),
-                buildPluginInfo(plugins)
+                buildPluginInfo(plugins),
+                buildOsInfo()
         );
     }
 
+    private Os buildOsInfo() {
+        org.jsoftbiz.utils.OS os = org.jsoftbiz.utils.OS.getOs();
```

Comment:
As far as I see all relevant error cases are handled in that class so that no exception should be thrown. If the code is being run on a yet unknown OS, the structure will be populated with the contents of the `os.name`, `os.version` and `os.arch` system properties.